presto-product-tests/src/main/java/com/facebook/presto/tests/hive/TestHiveCoercion.java

```diff
@@ -0,0 +1,354 @@
+            .setNoData()
+            .build();
+
+    @TableDefinitionsRepository.RepositoryTableDefinition
+    public static final HiveTableDefinition HIVE_COERCION_RCBINARY = tableDefinitionBuilder("RCFILE", Optional.of("RCBINARY"), Optional.of("SERDE 'org.apache.hadoop.hive.serde2.columnar.LazyBinaryColumnarSerDe'"))
+            .setNoData()
+            .build();
+
+    private static HiveTableDefinition.HiveTableDefinitionBuilder tableDefinitionBuilder(String fileFormat, Optional<String> recommendTableName, Optional<String> rowFormat)
+    {
+        String tableName = format(tableNameFormat, (recommendTableName.isPresent() ? recommendTableName.get() : fileFormat).toLowerCase(Locale.ENGLISH));
+        return HiveTableDefinition.builder(tableName)
+                .setCreateTableDDLTemplate("" +
+                        "CREATE TABLE %NAME%(" +
+                        "    tinyint_to_smallint        TINYINT," +
+                        "    tinyint_to_int             TINYINT," +
+                        "    tinyint_to_bigint          TINYINT," +
+                        "    smallint_to_int            SMALLINT," +
+                        "    smallint_to_bigint         SMALLINT," +
+                        "    int_to_bigint              INT," +
+                        "    bigint_to_varchar          BIGINT," +
+                        "    varchar_to_bigint          STRING" +
+                        ") " +
+                        "PARTITIONED BY (id BIGINT) " +
+                        (rowFormat.isPresent() ? "ROW FORMAT " + rowFormat.get() + " " : " ") +
+                        "STORED AS " + fileFormat);
+    }
+
+    private static HiveTableDefinition.HiveTableDefinitionBuilder parquetTableDefinitionBuilder()
+    {
+        return HiveTableDefinition.builder("parquet_hive_coercion")
+                .setCreateTableDDLTemplate("" +
+                        "CREATE TABLE %NAME%(" +
+                        "    tinyint_to_smallint        TINYINT," +
+                        "    tinyint_to_int             TINYINT," +
+                        "    tinyint_to_bigint          TINYINT," +
+                        "    smallint_to_int            SMALLINT," +
+                        "    smallint_to_bigint         SMALLINT," +
+                        "    int_to_bigint              INT," +
+                        "    bigint_to_varchar          BIGINT," +
+                        "    varchar_to_bigint          STRING" +
+                        ") " +
+                        "PARTITIONED BY (id BIGINT) " +
+                        "STORED AS PARQUET");
+    }
+
+    private static HiveDataSource getTextFileDataSource()
+    {
+        return createResourceDataSource(format(tableNameFormat, "textfile"), String.valueOf(ThreadLocalRandom.current().nextLong(Long.MAX_VALUE)), "com/facebook/presto/tests/hive/data/all_types/data.textfile");
```

Comment:
How's that work if `data.textfile` contains data of different types and different schema?
Namely, it has `tinyint, smallint, int, bigint, float/real, double, decimal, decimal(10,5), timestamp, date, string, varchar(10), char(10), boolean and binary` while here you need `tinyint, tinyint, tinyint, smallint, smallint, int, bigint, string`.
I don't think we want to introduce this implicit coercion for smallint -> tinyint, tinyint -> int, float -> smallint and so on.
Do I miss something?

I think you should introduce separate textfile for your tests to the resources.